worker-workflow/src/main/resources/workflow-control.js

```diff
@@ -385,3 +385,17 @@ function fieldHasStringValue(document, fieldName, value) {
 
     return false;
 }
+
+function fieldHasAnyStringValue(document, fieldName, values) {
+    var valueFound = false;
+
+    if(fieldValues)
+    {
+        values.forEach(currentValue => {
+            if(fieldHasStringValue(fieldValues, currentValue)) {
```

Comment:
This means that you are calling `document.getField(fieldName).getValues();` repeatedly.  Re-work this to not use `fieldHasStringValue`.

Reply:
@dermot-hardy I have brought out the actual value comparison into a new function so it can also be used for fieldHasAnyStringValue?
Added couple of tests to ensure fieldHasStringValue still works as expected